miden-lib/build.rs

```diff
@@ -303,35 +308,53 @@ fn compile_note_scripts(source_dir: &Path, target_dir: &Path, assembler: Assembl
     Ok(())
 }
 
-// COMPILE DEFAULT ACCOUNT COMPONENTS
+// COMPILE ACCOUNT COMPONENTS
 // ================================================================================================
 
-const BASIC_WALLET_CODE: &str = "
-    export.::miden::contracts::wallets::basic::receive_asset
-    export.::miden::contracts::wallets::basic::create_note
-    export.::miden::contracts::wallets::basic::move_asset_to_note
-";
-
-const RPO_FALCON_AUTH_CODE: &str = "
-    export.::miden::contracts::auth::basic::auth_tx_rpo_falcon512
-";
-
-const BASIC_FUNGIBLE_FAUCET_CODE: &str = "
-    export.::miden::contracts::faucets::basic_fungible::distribute
-    export.::miden::contracts::faucets::basic_fungible::burn
-";
-
-/// Compiles the default account components into a MASL library and stores the complied files in
-/// `target_dir`.
-fn compile_account_components(target_dir: &Path, assembler: Assembler) -> Result<()> {
-    for (component_name, component_code) in [
-        ("basic_wallet", BASIC_WALLET_CODE),
-        ("rpo_falcon_512", RPO_FALCON_AUTH_CODE),
-        ("basic_fungible_faucet", BASIC_FUNGIBLE_FAUCET_CODE),
-    ] {
-        let component_library = assembler.clone().assemble_library([component_code])?;
+/// Compiles the account components into a MASL library and stores the complied files in
+/// `target_dir`. Those libraries have `account_components` as their namespace and the name of the
+/// file as the module name.
+///
+/// For example, a `basic_wallet.masm` file could be imported as
+/// `use.account_components::basic_wallet`.
```

Comment:
This is a bit counter-intuitive, but it actually doesn't matter what the namespace is and we probably shouldn't rely on it this way. Since these just re-export procedures from `miden-lib`, these procedures can still be accessed via `miden-lib` paths - and we want users to do that anyway (e.g., via `miden::contracts::wallets::basic` namespace).

As mentioned in one of the previous comments, we probably don't want users to import these libraries into the assembler directly - but rather go though `miden-lib` (which is automatically imported by `TransactionKernel` assembler)

Reply:
Right, that makes perfect sense, I forgot users were still requird to import `miden-lib` anyway.

I reverted the library-related changes so this PR only effectively changes the build process.

Updated each component's docs to add a hint for how to link against these components.

Also removed the changelog entry and added the `no-changelog` label since the PR no longer changes anything about the functionality.